File: camerax_platform_interface/lib/src/core/camera_control.dart
import 'package:camerax_platform_interface/src/core.dart';

abstract interface class CameraControl {
  Future<void> cancelFocusAndMetering();
  Future<void> enableLowLightBoostAsync(bool lowLightBoost);
  Future<void> enableTorch(bool torch);
  Future<int> setExposureCompensationIndex(int value);
  Future<void> setLinearZoom(double linearZoom);
  Future<void> setTorchStrengthLevel(int torchStrengthLevel);
  Future<void> setZoomRatio(double ratio);
  Future<FocusMeteringResult> startFocusAndMetering(FocusMeteringAction action);
}

File: camerax_ios/ios/Classes/core/CameraControl.swift
//
//  CameraControl.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/5/22.
//

import Foundation

public protocol CameraControl: NSObjectProtocol {
    func enableTorch(_ torch: Bool) throws -> Void
    func setZoomRatio(_ ratio: CGFloat) throws -> Void
    func setLinearZoom(_ linearZoom: CGFloat) throws -> Void
    func startFocusAndMetering(_ action: FocusMeteringAction) throws -> FocusMeteringResult
    func cancelFocusAndMeteriing() throws -> Void
    func setExposureCompensationIndex(_ value: Int) throws -> Int
}
